### ISSUE TITLE
[Discuss] Shift simulators to use pandas?

### ISSUE THREAD

**Issue by 95-martin-orion** [quantumlib/Cirq#3708]
**Is your feature request related to a use case or problem? Please describe.**

Raised in #3698. This issue is just to capture discussion; we do not need to act on this unless a decision is reached.

Simulator sweeps can potentially return large collections of results, which are only organized as an ordered list. For more complex results (e.g. lists of expectation values), this might not be the clearest presentation.

**Describe the solution you'd like**

Results could be packaged in a `pandas.DataFrame` object, similar to the `Sampler` API. This would be a breaking change, as both internal and external simulators rely on the existing (non-pandas) API

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by balopat** [comment]
+1 to make the gap smaller between Sampler and Simulator

**Comment by 95-martin-orion** [comment]
This is best described as a component of #3868 - support for multiple measurement-result formats will narrow the gap between the Sampler and Simulator APIs.

**Comment by 95-martin-orion** [comment]
Duplicate of #3868

**Comment by tanujkhattar** [comment]
Closing as marked duplicate by @95-martin-orion in the above comment.
